qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -617,24 +659,41 @@ def run_experiment(self, experiment: QasmQobjExperiment) -> dict[str, ...]:
                             value >>= 1
                         if value != int(operation.conditional.val, 16):
                             continue
-                # Check if single  gate
                 if operation.name == "unitary":
                     qubits = operation.qubits
                     gate = operation.params[0]
                     self._add_unitary(gate, qubits)
+                elif operation.name in ("id", "u0", "delay"):
```

Comment:
I'm not fully sure that `delay` is the same as `id` in `BasicSimulator`

Reply:
@jakelishman what do you think? Should be `delay` be `id`, or warning, or not accepted?

Reply:
The basic simulator is noiseless, and in noiseless quantum-state evolution a time-delay is the identity op. I think it's fine / correct to accept delay and do nothing.